src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs

```diff
@@ -286,9 +286,11 @@ private DynValue Processing_Loop(int instructionPtr, bool canAwait = false)
 							ref var tab = ref m_ValueStack.Peek();
 							if (top.Type != DataType.Table) throw new InternalErrorException("v-stack top NOT table");
 							if (tab.Type != DataType.Table) throw new InternalErrorException("v-stack tab NOT table");
-							if (top.Table.ModifierFlags.HasFlag(MemberModifierFlags.Static))
+							
+							// NumVal > 0 = whether base class is present
+							if (i.NumVal > 0 && (top.Table.ModifierFlags.HasFlag(MemberModifierFlags.Static) || top.Table.ModifierFlags.HasFlag(MemberModifierFlags.Sealed)))
 							{
-								throw ScriptRuntimeException.BaseInvalidModifier(MemberModifierFlags.Static.ToString().ToLowerInvariant(), top.Table.Get("Name").String ?? "(null)", currentFrame.Function.strings[i.NumVal]);
+								throw ScriptRuntimeException.BaseInvalidModifier(top.Table.ModifierFlags.HasFlag(MemberModifierFlags.Static) ? MemberModifierFlags.Static.ToString().ToLowerInvariant() : MemberModifierFlags.Sealed.ToString().ToLowerInvariant(), top.Table.Get("Name").String ?? "(null)", currentFrame.Function.strings[i.NumVal]);
```

Comment:
This is a spaghetti statement. Split into two separate if statements and just use "static" and "sealed" strings.

Reply:
Fixed in 8462beb